st2common/st2common/util/virtualenvs.py

```diff
@@ -182,13 +185,33 @@ def remove_virtualenv(virtualenv_path, logger=None):
     return True
 
 
-def install_requirements(virtualenv_path, requirements_file_path):
+def install_requirements(virtualenv_path, requirements_file_path, proxy_config=None, logger=None):
     """
     Install requirements from a file.
     """
+    logger = logger or LOG
     pip_path = os.path.join(virtualenv_path, 'bin/pip')
-    cmd = [pip_path, 'install', '-U', '-r', requirements_file_path]
+    cmd = [pip_path]
+
+    if proxy_config:
+        cert = proxy_config.get('proxy_ca_bundle_path', None)
+        https_proxy = proxy_config.get('https_proxy', None)
+        http_proxy = proxy_config.get('http_proxy', None)
+
+        if http_proxy:
+            cmd.extend(['--proxy', http_proxy])
+
+        if https_proxy:
```

Comment:
A bit confused - what happens if both - http_proxy and https_proxy parameters are specified?

It looks like `--proxy` command line argument will get passed to `pip` twice, will this work? Or will just the last entry (https_proxy) win?